govcd/network_pool.go

```diff
@@ -102,3 +102,271 @@ func (vcdClient *VCDClient) GetNetworkPoolByName(name string) (*NetworkPool, err
+// It can create any type of network pool
+func (vcdClient *VCDClient) CreateNetworkPool(config *types.NetworkPool) (*NetworkPool, error) {
+	client := vcdClient.Client
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointNetworkPools
+	apiVersion, err := client.getOpenApiHighestElevatedVersion(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	urlRef, err := client.OpenApiBuildEndpoint(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	result := &NetworkPool{
+		NetworkPool: &types.NetworkPool{},
+		vcdClient:   vcdClient,
+	}
+
+	err = client.OpenApiPostItem(apiVersion, urlRef, nil, config, result.NetworkPool, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return result, nil
+}
+
+// Update will change all changeable network pool items
+func (np *NetworkPool) Update() error {
+	if np == nil || np.NetworkPool == nil || np.NetworkPool.Id == "" {
+		return fmt.Errorf("network pool must have ID")
+	}
+	if np.vcdClient == nil || np.vcdClient.Client.APIVersion == "" {
+		return fmt.Errorf("network pool '%s': no client found", np.NetworkPool.Name)
+	}
+
+	client := np.vcdClient.Client
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointNetworkPools
+	apiVersion, err := client.getOpenApiHighestElevatedVersion(endpoint)
+	if err != nil {
+		return err
+	}
+
+	urlRef, err := client.OpenApiBuildEndpoint(endpoint, np.NetworkPool.Id)
+	if err != nil {
+		return err
+	}
+
+	err = client.OpenApiPutItem(apiVersion, urlRef, nil, np.NetworkPool, np.NetworkPool, nil)
+	if err != nil {
+		return err
+	}
+
+	if err != nil {
+		return fmt.Errorf("error updating network pool '%s': %s", np.NetworkPool.Name, err)
+	}
+
+	return nil
+}
+
+// Delete removes a network pool
+func (np *NetworkPool) Delete() error {
+	if np == nil || np.NetworkPool == nil || np.NetworkPool.Id == "" {
+		return fmt.Errorf("network pool must have ID")
+	}
+	if np.vcdClient == nil || np.vcdClient.Client.APIVersion == "" {
+		return fmt.Errorf("network pool '%s': no client found", np.NetworkPool.Name)
+	}
+
+	client := np.vcdClient.Client
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointNetworkPools
+	apiVersion, err := client.getOpenApiHighestElevatedVersion(endpoint)
+	if err != nil {
+		return err
+	}
+
+	urlRef, err := client.OpenApiBuildEndpoint(endpoint, np.NetworkPool.Id)
+	if err != nil {
+		return err
+	}
+
+	err = client.OpenApiDeleteItem(apiVersion, urlRef, nil, nil)
+	if err != nil {
+		return err
+	}
+
+	if err != nil {
+		return fmt.Errorf("error deleting network pool '%s': %s", np.NetworkPool.Name, err)
+	}
+
+	return nil
+}
+
+// CreateNetworkPoolGeneve creates a network pool of GENEVE type
+// The function retrieves the given NSX-T manager and corresponding transport zone names
+// If the transport zone name is empty, the first available will be used
+func (vcdClient *VCDClient) CreateNetworkPoolGeneve(name, description, managerName, transportZoneName string) (*NetworkPool, error) {
```

Comment:
IMO this adds immediate clarity for what kind of "managerName" it is
```suggestion
func (vcdClient *VCDClient) CreateNetworkPoolGeneve(name, description, nsxtManagerName, transportZoneName string) (*NetworkPool, error) {
```